src/plugins/data/common/index_patterns/index_patterns/index_pattern.ts

```diff
@@ -195,14 +204,105 @@ export class IndexPattern implements IIndexPattern {
       };
     });
 
+    const runtimeFields = {
+      ...this.getComputedRuntimeFields(),
+      ...this.getComputedRuntimeComposites(),
+    };
+
     return {
       storedFields: ['*'],
       scriptFields,
       docvalueFields,
-      runtimeFields: this.runtimeFieldMap,
+      runtimeFields,
     };
   }
 
+  /**
+   * Method to aggregate all the runtime fields which are **not** created
+   * from a parent composite runtime field.
+   * @returns A map of runtime fields
+   */
+  private getComputedRuntimeFields(): Record<string, RuntimeField> {
+    return Object.entries(this.runtimeFieldMap).reduce((acc, [name, field]) => {
+      const { type, script, parentComposite } = field;
+
+      if (parentComposite !== undefined) {
+        return acc;
+      }
+
+      const runtimeFieldRequest: RuntimeField = {
+        type,
+        script,
+      };
+
+      return {
+        ...acc,
+        [name]: runtimeFieldRequest,
+      };
+    }, {});
+  }
+
+  /**
+   * This method reads all the runtime composite fields
+   * and aggregate the subFields
+   *
+   * {
+   *   "compositeName": {
+   *     "type": "composite",
+   *     "script": "emit(...)" // script that emits multiple values
+   *     "fields": {  // map of subFields available in the Query
+   *       "field_1": {
+   *         "type": "keyword"
+   *       },
+   *       "field_2": {
+   *         "type": "ip"
+   *       },
+   *     }
+   *   }
+   * }
+   *
+   * @returns A map of runtime fields
+   */
+  private getComputedRuntimeComposites(): Record<string, RuntimeField> {
```

Comment:
That could work. I am not sure if the benefit of extracting to a new file outweighs the indirection created but I am happy either ways. For the tests, you know how I prefer to test the integration (in this case calling `getComputedFields()`) instead of unit testing. We'll start first with the integration and add unit tests if needed.